### PR TITLE
Add tests about generic list of submodels

### PR DESCRIPTION
**Pull request by GuillaumeDesforges**
I'm struggling with some issues with a generic model with a list of elements of submodels.
As I'm trying to solve my own problem, I wrote tests to see if the issue was coming from the library.

Turns out it is not, but I figure it's always useful to add tests.

### CODE REVIEW

**Comment by dapper91** [comment]
@GuillaumeDesforges please, change target branch from master to dev

File: tests/test_generics.py [hunk 1]
```diff
@@ -70,6 +70,60 @@ class TestModel(BaseXmlModel, tag='model1'):
     assert_xml_equal(actual_xml, xml)
 
 
+def test_generic_list():
+    GenericType = TypeVar('GenericType')
+
+    class GenericModel(BaseGenericXmlModel, Generic[GenericType], tag="model1"):
+        elems: list[GenericType] = element(tag="elem")
```
Comment (dapper91) [inline tests/test_generics.py:77]: `list[...]` syntax doesn't work in python < 3.9, lets change it to `List[GenericType]`

Comment (GuillaumeDesforges) [reply tests/test_generics.py]: Fixed in [cde29c2](https://github.com/dapper91/pydantic-xml/pull/66/commits/cde29c27f663bb5c48bae30e83079b504bb1f5be)

**Comment by dapper91** [comment]
@GuillaumeDesforges please, change target branch from master to dev

**Comment by GuillaumeDesforges** [comment]
Sure, sorry about that.

**Pull request merged**